### ISSUE TITLE
TransformerEncoder/TransformerDecoder do not propagate keras masks

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#137]
If you have a series of encoder blocks, the first one will correct pass a mask tensor to the MHA layer, but the mask will not propagate the mask to the next encoder block.

Ran into this when trying to just pool the output of a transformer, we end up just pooling every output position including padding.

Repro:
https://colab.sandbox.google.com/gist/mattdangerw/9f6eb19fbc81729d749101f967005043/transformerencoder-mask-bug.ipynb

**Comment by mattdangerw** [comment]
@chenmoneygithub I think we just need to propagate the mask, but can you take a look and make sure I am not missing anything here?

**Comment by chenmoneygithub** [comment]
I see, we should probably just explicitly set `output._keras_mask = previous mask`.

**Comment by mattdangerw** [comment]
I think we would just do this as `compute_mask`?

https://www.tensorflow.org/guide/keras/masking_and_padding#supporting_masking_in_your_custom_layers

**Comment by mattdangerw** [comment]
Oops sorry it's even simpler than that. We just need `self.supports_masking = True` in init and a unit test I think.

https://www.tensorflow.org/guide/keras/masking_and_padding#opting-in_to_mask_propagation_on_compatible_layers

**Comment by mattdangerw** [comment]
Fixed on https://github.com/keras-team/keras-nlp/pull/139
